src/main/java/seedu/address/logic/commands/ScheduleCommand.java

```diff
@@ -0,0 +1,171 @@
+        return new CommandResult(String.format(MESSAGE_SCHEDULE_ASSIGNMENT_SUCCESS, scheduledAssignment));
+    }
+
+    private Schedule createValidSchedule(Assignment assignmentToSchedule, List<Task> taskList) throws CommandException {
+        LocalDateTime newDeadline = doBefore.toLocalDateTime();
+        if (assignmentToSchedule.getDeadline().isBefore(doBefore)) {
+            newDeadline = assignmentToSchedule.getDeadline().toLocalDateTime();
+        }
+
+        LocalDateTime nearestTime = roundToHour(LocalDateTime.now().plusHours(1));
+
+        List<LocalDateTime> possibleTime = generateAllPossibleTime(nearestTime,
+                newDeadline, taskList);
+
+        if (possibleTime.isEmpty()) {
+            throw new CommandException("No possible schedule");
+        }
+        return getRandom(possibleTime);
+    }
+
+    private LocalDateTime roundToHour(LocalDateTime time) {
+        int min = time.getMinute();
+        time = time.minusMinutes(min % 60);
+        return time;
+    }
+
+    private Schedule getRandom(List<LocalDateTime> list) {
+        int rnd = new Random().nextInt(list.size());
+        Deadline suggestedStartTime = new Deadline(list.get(rnd));
+        Deadline suggestedEndTime = new Deadline(list.get(rnd).plusHours(expectedTime));
+        return new Schedule(suggestedStartTime, suggestedEndTime);
+    }
+
+    private List<LocalDateTime> generateAllPossibleTime(LocalDateTime start, LocalDateTime end, List<Task> taskList) {
+        List<LocalDateTime> possibleTime = new ArrayList<>();
+        for (LocalDateTime i = start; i.isBefore(end); i = i.plusHours(1)) {
+            boolean can = true;
+            // working hours
+            if (!isWorkingHour(i, i.plusHours(expectedTime))) {
+                can = false;
+            }
+            // no overlap
+            for (Task j: taskList) {
+                if (overlap(i, i.plusHours(expectedTime), j)) {
+                    can = false;
+                }
+            }
+
+            if (can) {
+                possibleTime.add(i);
+            }
+        }
+        return possibleTime;
+    }
+
+    private boolean isWorkingHour(LocalDateTime start, LocalDateTime end) {
+        return !start.toLocalTime().isBefore(WORKING_START_TIME) && !end.toLocalTime().isAfter(WORKING_END_TIME);
+    }
+
+    private boolean overlap(LocalDateTime start, LocalDateTime end, Task task) {
+        if (task instanceof Assignment) {
+            if (!((Assignment) task).getSchedule().isScheduled()) {
+                return false;
+            }
+            return (!start.isBefore(((Assignment) task).getSchedule().getSuggestedStartTime().toLocalDateTime())
+                    && !end.isAfter(((Assignment) task).getSchedule().getSuggestedEndTime().toLocalDateTime()));
+        }
+        return (!start.isBefore(task.getTime().toLocalDateTime())
+                && !end.isAfter(((Lesson) task).getEndTime().toLocalDateTime()));
+    }
```

Comment:
According to CS2103T conventions, boolean methods and variablels should be named to sound like a boolean (i.e haveOverlap, isOverlappingTime etc)